src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -338,29 +315,157 @@ public void storeContext(Context context, HttpServletRequest request) {
+	}
+
+	public int getParentAccount(int accountId){
+		M_log.debug("getParentAccount() call");
+		// https://umich.test.instructure.com:443/api/v1/accounts/78
+		String url = makeCanvasURL("/accounts/" + accountId);
+		HttpResponse response = processApiCall(new HttpGet(url));
+		if (response == null) {
+			M_log.error(String.format("getting parent for account %s for SubAccountAdmin check failed due " +
+					"to null response", accountId));
+			return 0;
+		}
+		int statusCode = response.getStatusLine().getStatusCode();
+		if (statusCode != HttpStatus.SC_OK) {
+			M_log.error(String.format("getting parent for account %s for SubAccountAdmin check failed " +
+					"with status %s and " + "with error message", accountId, statusCode, response.toString()));
+			return 0;
+		}
+
+		String parentAccountResponse;
+		int parentAccountId = 0;
+		String errMsg = "As part of SubAccountAdmin check, couldn't get " +
+				"the ParentId for account %s due to %s";
+		try {
+			parentAccountResponse = EntityUtils.toString(response.getEntity(), "UTF-8");
+			JSONObject parentAccountJson = new JSONObject(parentAccountResponse);
+			if (!parentAccountJson.isNull("parent_account_id")) {
+				parentAccountId = (int) parentAccountJson.get("parent_account_id");
+			}
+		} catch (IOException e) {
+			M_log.error(String.format(errMsg, accountId, e.getMessage()));
+			return 0;
+		} catch (Exception e) {
+			M_log.error(String.format(errMsg, accountId, e.getMessage()));
+			return 0;
+		}
+		return parentAccountId;
+
+	}
+
+	private int getCourseAccountId(String courseId){
+		M_log.debug("getCourseAccountId() call");
+		int course_account_id = 0;
+		String url = makeCanvasURL("/courses/" + courseId);
+		HttpResponse response = processApiCall(new HttpGet(url));
+		if (response == null) {
+			return 0;
+		}
+		int statusCode = response.getStatusLine().getStatusCode();
+		if (statusCode != HttpStatus.SC_OK) {
+			return course_account_id;
+		}
+		String courseData = null;
+		String errMsg = "As part of SubAccountAdmin check, couldn't get accountId the course %s belong to";
+		try {
+			courseData = EntityUtils.toString(response.getEntity(), "UTF-8");
+		} catch (IOException e) {
+			M_log.error(String.format(errMsg, courseId));
+			return 0;
+		} catch (Exception e) {
+			M_log.error(String.format(errMsg, courseId));
+			return 0;
+		}
+		JSONObject courseJson = new JSONObject(courseData);
+		if (!courseJson.isNull("account_id")) {
+			course_account_id = (int) courseJson.get("account_id");
+		}
+		return course_account_id;
+	}
+
+	private String makeCanvasURL(String url){
+		url = canvasURL+Utils.CANVAS_API_VERSION+url;
+		M_log.info("API call "+url);
+		return url;
+	}
+
+	private void fillCcmValuesForContext(Map<String, String> ltiValues, HttpServletRequest request, TcSessionData tc) {
+		HashMap<String, Object> customValuesMap = tc.getCustomValuesMap();
+		ltiValues.put(Utils.LTI_PARAM_CANVAS_COURSE_ID, request.getParameter(Utils.LTI_PARAM_CANVAS_COURSE_ID));
```

Comment:
I modified it but i don't know if transfer method is really in use as all the values that goes in the context is now grabbed from the `TcSession` instead of `HttpServlet`